### PR TITLE
Add support for API 23+

### LINKED ISSUE #16: When running on an api 23 emulator and targeting api 23, permissions are not granted
Since `write_external_storage` is required, if an app compiles and targets api 23 it needs to accept the external storage permission. This can be done by installing with -g, however the screenshot tests don't do this so writing to external storage always fails.

Compiling against 23 and targeting 22 does not exhibit these symptoms as it automatically grants all requested permissions.

### PR DESCRIPTION
**Pull request by xiphirx**
Fixes #16 

`make integration-tests` passes on an API 25 emulator.

We should maybe track the status of https://issuetracker.google.com/issues/37077653 to introduce a "better" method of solving this issue in the future.

### CODE REVIEW

**Comment by sridoodla** [comment]
https://developer.android.com/reference/android/support/test/rule/GrantPermissionRule.html

Maybe that's a better way to go about it?

**Comment by xiphirx** [comment]
We would need to introduce the testing support library here, which is apparently still in beta. I don't know how updating affects the rest of the project, so I'd rather hold off on that for now.

File: core/src/main/java/com/facebook/testing/screenshot/internal/ScreenshotDirectories.java [hunk 1]
```diff
@@ -32,13 +44,41 @@ public File get(String type) {
...
   private void checkPermissions() {
-    int res = mContext.checkCallingOrSelfPermission("android.permission.WRITE_EXTERNAL_STORAGE");
-    if (res != PackageManager.PERMISSION_GRANTED) {
-      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-        throw new RuntimeException("This does not currently work on API 23+, see "
-            + "https://github.com/facebook/screenshot-tests-for-android/issues/16 for details.");
-      } else {
-        throw new RuntimeException("We need WRITE_EXTERNAL_STORAGE permission for screenshot tests");
+    for (String permission : REQUIRED_PERMISSIONS) {
+      if (mContext.checkCallingOrSelfPermission(permission) == PackageManager.PERMISSION_GRANTED) {
```
Comment (tdrhq) [inline core/src/main/java/com/facebook/testing/screenshot/internal/ScreenshotDirectories.java]: READ_EXTERNAL_STORAGE was added in API level 16, so this check should will fail always in < 16. Can you check this on ice cream sandwich?

Comment (xiphirx) [inline core/src/main/java/com/facebook/testing/screenshot/internal/ScreenshotDirectories.java]: I'm using a string constant for that exact reason :) Look at line 32

**Comment by xiphirx** [comment]
Friendly ping :)

**Comment by weitzj** [comment]
Hi. It would be nice, if you could release a new version of the plugin with this feature enabled.

**Pull request merged**